lib/app_web/controllers/api_controller.ex

```diff
@@ -0,0 +1,49 @@
+defmodule AppWeb.ApiController do
+  use AppWeb, :controller
+  import SweetXml
+
+  def create(conn, %{"image" => image}) do
+
+    # Check if file is an image
+    fileIsAnImage = String.contains?(image.content_type, "image")
+
+    if fileIsAnImage do
+
+      # Create `CID` from file contents
+      {:ok, file_binary} = File.read(image.path)
+      file_cid = Cid.cid(file_binary)
+      file_name = "#{file_cid}.#{Enum.at(MIME.extensions(image.content_type), 0)}"
+
+      # Upload to S3
+      upload = image.path
+      |> ExAws.S3.Upload.stream_file()
+      |> ExAws.S3.upload("imgup-original", file_name, acl: :public_read)
+      |> ExAws.request(get_ex_aws_request_config_override())
+
+      # Check if upload was successful
+      case upload do
+        {:ok, body} ->
+          url = body.body |> xpath(~x"//text()") |> List.to_string()
```

Comment:
Didn't know you spoke `xpath` ... 💭 
(where did you find this code? please link to source)

Reply:
or at least leave a comment explaining what this does. 🙏

Reply:
I just followed the guide from https://github.com/kbrw/sweet_xml. It made sense following the XML string that was returned from the `ex_aws` package.
I can add a comment to explain the README.